qemu/tests/virtio_chardev_trace.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from avocado.utils import process
+from virttest import error_context
+from virttest import env_process
+from virttest import data_dir
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    virtio-trace support testing:
+    1) Make FIFO per CPU in a host.
+    2) reboot guest with virtio-serial device, control path and data path per CPU.
+    3) Download trace agent and compile it.
+    4) Enable ftrace in the guest.
+    5) Run trace agent in the guest.
+    6) Open FIFO in a host.
+    7) Start to read trace data by ordering from a host
+    8) Stop to read trace data by ordering from a host
+    9) repeat 7) and 8) with different CPU
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def execute_host_cmd(host_cmd, timeout=60):
+        """
+        Execute the host_cmd on host, limited in timeout period
+
+        param host_cmd: the host_cmd to run on host
+        param timeout: the timeout for running this command
+        return: the output of the host_cmd
+        """
+        logging.info("Executing host command: %s" % host_cmd)
+        cmd_result = process.getstatusoutput(host_cmd, timeout=timeout, shell=True)
+        return cmd_result
+
+    nums_cpu = int(params.get("smp", 1))
+    serials = params.get("serials", '')
+    v_path = "/tmp/virtio-trace/"
+    execute_host_cmd("mkdir {}".format(v_path))
+    for t in ["in", "out"]:
+        execute_host_cmd("mkfifo /tmp/virtio-trace/agent-ctl-path.{}".format(t))
+        for x in range(int(nums_cpu)):
+            execute_host_cmd("mkfifo {}trace-path-cpu{}.{}".format(v_path, x, t))
+
+    for x in range(int(nums_cpu)):
+        serials += ' vs{} '.format(x+1)
+        params['serial_type_vs{}'.format(x+1)] = 'virtserialport'
+        params['chardev_backend_vs{}'.format(x+1)] = 'pipe'
+        params['serial_name_vs{}'.format(x+1)] = "trace-path-cpu{}".format(x)
+        params['serial_filename_vs{}'.format(x+1)] = "/tmp/virtio-trace/trace-path-cpu{}".format(x)
+    params['serials'] = serials
+    params['start_vm'] = "yes"
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    session.cmd_status_output("echo 1 > /sys/kernel/debug/tracing/events/sched/enable")
+    vm.copy_files_to(data_dir.get_deps_dir("virtio-trace"), '/home/')
+    session.cmd_status_output("/home/virtio-trace/trace-agent -o &")
```

Comment:
Please make you code more readable. explain what does `/home/virtio-trace/trace-agent -o &` used for.
```
guest_dst_path = '/home'
vm.copy_files_to(data_dir.get_deps_dir("virtio-trace"), guest_dst_path)
guest_trace_agent_cmd = "/home/virtio-trace/trace-agent"
session.cmd_status_output(guest_trace_agent_cmd)
```